gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/versions/BaselineVersions.java

```diff
@@ -162,4 +166,28 @@ private static File rootVersionsPropsFile(Project project) {
                 })
                 .collect(Collectors.toSet());
     }
+
+    /**
+     * Compares {@code versions.props} matchers by weight. Higher weight means the matcher is more specific.
+     * For example,
+     * <pre>
+     *     com.google.guava:guava
+     * </pre>
+     * is more specific than
+     * <pre>
+     *     com.google.guava:*
+     * </pre>
+     */
+    static final Comparator<String> VERSIONS_PROPS_ENTRY_SPECIFIC_COMPARATOR =
+            Comparator.comparing(BaselineVersions::versionsPropsMatcherWeight);
+
+    /**
+     * The weight of a matcher in {@code versions.props} according to the disambiguation logic defined in
+     * {@code nebula.dependency-recommender}.
+     *
+     * This matches the logic in {@link FuzzyVersionResolver}.
+     */
+    private static int versionsPropsMatcherWeight(String propertiesLine) {
+        return Streams.stream(Splitter.on("*").split(propertiesLine)).mapToInt(String::length).sum();
```

Comment:
nit: `CharMatcher.isNot('*').countIn(propertiesLine)`

Reply:
your guava fu is too high